ibm_ace/datadog_checks/ibm_ace/resources.py

```diff
@@ -0,0 +1,208 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base.utils.common import total_time_to_temporal_percent
+from datadog_checks.base.utils.constants import SECOND
+
+
+class Resource:
+    def __init__(self, name):
+        self.name = name.replace('.', '_')
+
+    def normalized_metric_name(self, metric):
+        # https://www.ibm.com/support/pages/apar/IT39095
+        if ' ' in metric:
+            metric = metric.title().replace(' ', '')
+
+        return metric.replace('-', '_')
+
+    def full_metric_name(self, metric):
+        return f'{self.name}.{self.normalized_metric_name(metric)}'
+
+    def parse_tags(self, global_tags, metric_data):
+        group = metric_data.pop('name')
+        return [f'group:{group}', *global_tags]
+
+    def submit(self, check, metric, value, tags):
+        # Most metrics seem to be counters:
+        # https://www.ibm.com/docs/en/app-connect/12.0?topic=performance-resource-statistics-data
+        check.monotonic_count(self.full_metric_name(metric), value, tags=tags)
+
+
+class ConnectDirectResource(Resource):
+    """
+    https://www.ibm.com/docs/en/app-connect/12.0?topic=data-connect-direct
+    """
+
+    def parse_tags(self, global_tags, metric_data):
+        tags = super().parse_tags(global_tags, metric_data)
+        metric_data.pop('ConnectionDetails', None)
+        return tags
+
+
+class FTPResource(Resource):
+    """
+    https://www.ibm.com/docs/en/app-connect/12.0?topic=data-ftp
+    """
+
+    def parse_tags(self, global_tags, metric_data):
+        tags = super().parse_tags(global_tags, metric_data)
+
+        ftp_protocol = metric_data.pop('Protocol', '')
+        if ftp_protocol:
+            tags.append(f'ftp_protocol:{ftp_protocol}')
+
+        return tags
+
+
+class JDBCConnectionPoolsResource(Resource):
+    """
+    https://www.ibm.com/docs/en/app-connect/12.0?topic=data-jdbc-connection-pools
+    """
+
+    def parse_tags(self, global_tags, metric_data):
+        tags = super().parse_tags(global_tags, metric_data)
+
+        jdbc_provider = metric_data.pop('NameOfJDBCProvider', '')
+        if jdbc_provider:
+            tags.append(f'jdbc_provider:{jdbc_provider}')
+
+        return tags
+
+    def submit(self, check, metric, value, tags):
+        if metric in ('ActualSizeOfPool', 'MaxDelayInMilliseconds', 'MaxSizeOfPool'):
```

Comment:
It would be nice to have logic abstracted for the exceptions to metric types, but I'm not sure of a great way to do this